src/preprocessing/passes/bv_ackermann.cpp

```diff
@@ -41,11 +42,57 @@ namespace passes {
 namespace
 {
 
-void storeFunction(
-    TNode func,
-    TNode term,
-    FunctionToArgsMap& fun_to_args,
-    SubstitutionMap& fun_to_skolem)
+void addLemmaForPair(TNode args1,
+                     TNode args2,
+                     const TNode func,
+                     AssertionPipeline* assertionsToPreprocess,
+                     NodeManager* nm)
+{
+  Node args_eq;
+
+  if (args1.getKind() == kind::APPLY_UF)
+  {
+    AlwaysAssert(args1.getOperator() == func);
+    AlwaysAssert(args2.getKind() == kind::APPLY_UF
+                 && args2.getOperator() == func);
+    AlwaysAssert(args1.getNumChildren() == args2.getNumChildren());
```

Comment:
Why are these AlwaysAssert?
Note that AlwaysAssert is always enabled (as thte name implies), i.e., not only for debug but also for production builds. An assertion should only be an AlwaysAssert, if it is a critical runtime error (abort conditions such as ID overflows, mem out, ...). It seems to me that these should be simple Asserts.